lib/API/JSON/ResponseBuilders/NewslettersResponseBuilder.php

```diff
@@ -27,12 +44,36 @@ public function build(NewsletterEntity $newsletter) {
       'updated_at' => $newsletter->getUpdatedAt()->format(self::DATE_FORMAT),
       'deleted_at' => ($deletedAt = $newsletter->getDeletedAt()) ? $deletedAt->format(self::DATE_FORMAT) : null,
       'parent_id' => ($parent = $newsletter->getParent()) ? $parent->getId() : null,
-      'segments' => $this->buildSegments($newsletter),
-      'options' => $this->buildOptions($newsletter),
-      'queue' => ($queue = $newsletter->getLatestQueue()) ? $this->buildQueue($queue) : false, // false for BC
       'unsubscribe_token' => $newsletter->getUnsubscribeToken(),
       'ga_campaign' => $newsletter->getGaCampaign(),
     ];
+
+    foreach ($relations as $relation) {
+      if ($relation === self::RELATION_QUEUE) {
+        $data['queue'] = ($queue = $newsletter->getLatestQueue()) ? $this->buildQueue($queue) : false; // false for BC
+      }
+      if ($relation === self::RELATION_SEGMENTS) {
+        $data['segments'] = $this->buildSegments($newsletter);
+      }
+      if ($relation === self::RELATION_OPTIONS) {
+        $data['options'] = $this->buildOptions($newsletter);
+      }
+      if ($relation === self::RELATION_TOTAL_SENT) {
+        $data['totalSent'] = $this->newslettersStatsRepository->getTotalSentCount($newsletter);
+      }
+      if ($relation === self::RELATION_CHILDREN_COUNT) {
+        $data['childrenCount'] = count($newsletter->getChildren());
```

Comment:
This will require to fetch and hydrate all children entities. What about creating a getChildrenCount() method which would only query DB for the count on newslettersRepository?